Orange/widgets/widget.py

```diff
@@ -145,6 +145,18 @@ def __new__(cls, parent=None, *args, **kwargs):
         if self.want_basic_layout:
             self.insertLayout()
 
+        def patched__init__(*args, _old__init__=cls.__init__, **kwargs):
+            """
+            After initialization, test input signal handlers with None as input
+            => catch connection deleting bugs early.
+            """
+            _old__init__(*args, **kwargs)
+            for input in self.inputs:
+                args = (None,) if input.single else (None, 0)
+                getattr(self, input.handler)(*args)
+
+        cls.__init__ = patched__init__
+
         return self
```

Comment:
No.

You are capturing the `cls.__init__` on every `__new__` call, but  the old `__init__` will (except for the first call) be a `patched__init__` of the penultimate widget instance. Note that the `pathched__init__` captures the `self` instance.

In essence this creates a chain of all class instances which have ever been created in a process and clears all of their input whenever a new instance of the class is created.